packages/client-http/src/ApplyManager.ts

```diff
@@ -61,6 +64,11 @@ export class ApplyManager {
     if (this.flushTimeout) {
       clearTimeout(this.flushTimeout);
     }
-    this.flushTimeout = setTimeout(() => this.flush(), this.timeout);
+
+    if ((this.resolveTokenPending.get(resolveToken)?.length || 0) >= this.maxBufferSize) {
+      this.flush();
+    } else {
+      this.flushTimeout = setTimeout(() => this.flush(), this.timeout);
```

Comment:
Does this mean that we wait until the buffer is full, before sending `apply`? If so, I think it's safer to try send all the apply data more often to avoid cases where an half-full buffer will never get sent

Reply:
nope! This means that if the buffer is full it will call flush (i.e. send the apply event). If the buffer is not full it will wait to see if anything else can be put in the buffer before the timeout ends, then send. In other words, the buffer size will call flush, otherwise act as before :)